scripts/submit-analytics.js

```diff
@@ -0,0 +1,201 @@
+    return request;
+  }
+}
+
+/**
+ * Generate a hash value of data
+ * @param {*} data
+ * @returns SHA256 of data
+ */
+function sha256(data) {
+  let hash = require("crypto").createHash("sha256");
+  hash.update(data);
+  return hash.digest("hex");
+}
+
+/**
+ * Collect analytics data from the runtime system
+ * @param {Object} context The app's package.json parsed as an object
+ * @returns {Object} Analytics payload
+ */
+async function fetchPlatformData(context) {
+  const os = require("os");
+  const { machineId } = require("node-machine-id");
+
+  let identifier = await machineId();
+  if (!identifier) {
+    identifier = sha256("unknown");
+  }
+
+  let framework = "node.js";
+  let frameworkVersion = process.version;
+  let jsEngine = "v8";
+
+  if (context.dependencies && context.dependencies["react-native"]) {
+    framework = "react-native";
+    frameworkVersion = context.dependencies["react-native"];
+    try {
+      const podfile = fs.readFileSync("../../ios/Podfile", "utf8");
+      if (podfile.includes("hermes_enabled => true")) {
+        jsEngine = "hermes";
+      } else {
+        jsEngine = "jsc";
+      }
+    } catch (err) {
+      doLog(`Cannot read ios/Podfile: ${err}`);
+      jsEngine = "unknown";
+    }
+  }
+  if (context.dependencies && context.dependencies["electron"]) {
+    framework = "electron";
+    frameworkVersion = context.dependencies["electron"];
+  }
+
+  const payloads = {
+    webHook: {
+      event: "install",
+      properties: {
+        token: "aab85907a13e1ff44a95be539d9942a9",
+        "JS Analytics Version": 2,
+        distinct_id: identifier,
+        "Anonymized Machine Identifier": identifier,
+        "Anonymized Application ID": sha256(__dirname),
+        Binding: "javascript",
+        Version: context.version,
+        Language: "javascript",
+        Framework: framework,
+        "Framework Version": frameworkVersion,
+        "JavaScript Engine": jsEngine,
+        "Host OS Type": os.platform(),
+        "Host OS Version": os.release(),
+        "Node.js version": process.version,
+      },
+    },
+  };
+
+  return payloads;
+}
+
+function isAnalyticsDisabled() {
+  let isDisabled = false;
+
+  // NODE_ENV is commonly used by JavaScript framework
+  if ("NODE_ENV" in process.env) {
+    isDisabled |= process.env["NODE_ENV"] === "production" || process.env["NODE_ENV"] === "test";
+  }
+
+  // If the user has specifically opted-out or if we're running in a CI environment
+  isDisabled |= "REALM_DISABLE_ANALYTICS" in process.env || "CI" in process.env;
+
+  return isDisabled;
+}
+
+/**
+ * Send collected analytics data to Realm's servers over HTTPS
+ * @param  {Object} payload analytics info
+ */
+async function dispatchAnalytics(payload) {
+  const https = require("https");
+
+  return new Promise((resolve, reject) => {
+    const webhookRequest = new Webhook().buildRequest(payload);
```

Comment:
If my suggestion at the top is useful, then

```suggestion
    const webhookRequest = getAnalyticsRequestUrl(payload);
```